File: backend_server/models/notification_model.go
package models

type NotificationModel struct {
	Email       string `json:"email" bson:"email"`
	SenderPhone string `json:"sender_phone" bson:"sender_phone"`
	SenderName  string `json:"sender_name" bson:"sender_name"`
	Type        int    `json:"type" bson:"type"`
	Content     string `json:"content" bson:"content"`
	Timestamp   int64  `json:"timestamp" bson:"timestamp"`
}

File: car_pooling_app/lib/model/notification/notification_model.dart

// ignore_for_file: non_constant_identifier_names

import 'package:freezed_annotation/freezed_annotation.dart';

part 'notification_model.freezed.dart';
part 'notification_model.g.dart';

@freezed
class NotificationModel with _$NotificationModel {
  const factory NotificationModel({
    required String email,
    required String sender_phone,
    required String sender_name,
    required int type,
    required String content,
    required int timestamp,
  }) = _NotificationModel;

  factory NotificationModel.fromJson(Map<String, dynamic> json) =>
      _$NotificationModelFromJson(json);
}

